### DISCUSSION TITLE
G7 companion app issue

### DISCUSSION THREAD

**Discussion by annele373** [NightscoutFoundation/xDrip#2753] [Q&A]
Hi! 

I've set up xDrip to pull data via companion app (G7), but it only works when my phone's screen is on, am I missing some setting or is there a workaround for this? 

Attaching screenshot of how my current notifications look like. 

Thank you!
![xdrip_comp_screenshot](https://user-images.githubusercontent.com/129327758/228646026-0472e889-f301-4fe3-8b5e-ff19b88555c1.jpg)

EDIT: 

xDrip version: 344deb4-2022.11.16

![xdrip_BT_settings_1](https://user-images.githubusercontent.com/129327758/228656311-38768697-19d9-4ff6-9eb8-a9096af2dbe7.jpg)
![xdrip_BT_settings_2](https://user-images.githubusercontent.com/129327758/228656376-4621ddc3-94e5-4887-8d16-76b79be0109b.jpg)

**Comment by Navid200** [comment]
Open your Dexcom app and look at all settings.  Is there any setting related to notifications?

**Reply by annele373** [reply]
the only thing I can see is the quick glance, but I have it turned on, figured it wouldn't work otherwise. Other than that, I don't have any quiet modes enabled, but there aren't any other settings to fiddle with really, the app is quite minimalistic :/

**Reply by Navid200** [reply]
I'm just speculating here.  
Can you please have a look at Android settings and go to apps and find xDrip and go to Advanced and find Battery there.  Is battery optimization disabled?

**Reply by annele373** [reply]
It's unrestricted battery usage, I am now looking at lock screen notifications, but they are all enabled as well

**Reply by Navid200** [reply]
@jamorham 
I've tagged someone who knows more then all of us about this.

But, in the meantime, would you please:
1- Edit your first post and include what version of xDrip you are using.

 2- Check the following xDrip setting.
Settings &#8722;> Less common settings &#8722;> Bluetooth Settings &#8722;>

**Reply by annele373** [reply]
Thank you! Edits added!

**Comment by jamorham** [comment]
which phone model and android version is this?

**Reply by annele373** [reply]
Hi its a Samsung Galaxy s21 ultra 5g, android version 13

**Reply by jamorham** [reply]
There are two things you can try:
1) In xDrip settings search for `Companion Bluetooth` and try enabling that.
2) Look in the phone app settings for anything relating to power saving where apps are restricted from background activity. Check both for xDrip and the official app and make sure they are unrestricted. Often phones can have multiple places where there are these special manufacturer power saving options.

**Reply by Diabeticliberty** [reply]
Hello annele 373, having now completely lost all faith in Libre 2 I have a Dexcom G7 on the way to me that they have offered for a trial. I wish to continue to use XDrip+ as my primary source of diabetes management as frankly the app has been so good to me for many years. Could I ask if you have managed to resolve the issue that you have posted here and if so how you set the G7 up using companion?

**Reply by annele373** [reply]
Hey, the xdrip direcr integration with G7 is now out, so I am using that with no issues :) found the instructions for correct setup quickly by just googling so should not be an issue! Good luck!!
